Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabFocusHandler.h

```diff
@@ -65,6 +64,7 @@ namespace AzToolsFramework::Prefab
         PrefabFocusOperationResult FocusOnPathIndex(AzFramework::EntityContextId entityContextId, int index) override;
         PrefabFocusOperationResult SetOwningPrefabInstanceOpenState(AZ::EntityId entityId, bool openState) override;
         AZ::EntityId GetFocusedPrefabContainerEntityId(AzFramework::EntityContextId entityContextId) const override;
+        TemplateId GetFocusedPrefabTemplateId(AzFramework::EntityContextId entityContextId) const override;
```

Comment:
Would be good to include why the focus API changes in description.

In addition, should `GetFocusedPrefabInstance()` be moved to public interface as well?

Reply:
It is my understanding that the PrefabFocusPublicRequestBus should be callable via scripting. So, putting the GetFocusedPrefabInstance function there would violate that rule as it returns an optional instance reference that can't be easily read in python as far as I'm aware. @AMZN-daimini , correct me if I'm wrong.

As for moving GetFocusedPrefabTemplateId() to public interface, the reasoning is the same. It adds additional capability for scripts to query template id which they can use to query prefab system later if needed. Plus I didn't want the PrefabSaveLoadHandler to be using both the public and the nonpublic interfaces at the same time.

Reply:
Correct, the idea was to only use TemplateIds and InstanceReferences in the internal interfaces/buses, and use EntityIds in the public interfaces. This both to hide the Prefab internals from client systems, and to avoid having to expose those to Python scripting.